### ISSUE TITLE
Assumption that CUST_CD,AS_CD is primary key of CUST_ASP is incorrect

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#425]
_Imported from trac ticket 425.
Created by **kbower**
Opened in trac: Nov 2, 2010, 12:02:09 PM
Last modified in trac: Nov 5, 2010, 12:54:33 PM_
***
_by kbower:_
There are some customers who have more than one account with the same financer.

Example:
```
SQL> select CUST_CD,AS_CD, ACCT_CD from CUST_ASP where cust_cd = 'BROWD54328' and as_cd = 'GE_MONEY';

CUST_CD    AS_CD      ACCT_CD
---------- ---------- ------------------------------
BROWD54328 GE_MONEY   4444555566667777
BROWD54328 GE_MONEY   6012507800003162
BROWD54328 GE_MONEY   6012987300000416
BROWD54328 GE_MONEY   6012987300002451
BROWD54328 GE_MONEY   6012987300004167
BROWD54328 GE_MONEY   6012987300005416

6 rows selected.
```
Further, there are many unfinished and bad practices that need to be refactored in this code.  For example, this line will pull thousands of records (all GE_MONEY) from the database for the webserver to filter in python:
```python
accts = [account for account in financer.accounts if account.customerid == self.customerid]
```

**Comment by kbower** [comment]
_**kbower** trac comment on Nov 2, 2010, 12:09:22 PM:_

I can have both a closed and an open account with GE_MONEY

**Comment by kbower** [comment]
_**kbower** trac comment on Nov 2, 2010, 2:36:08 PM:_

(In [[883]](../commit/4daf7ded1a8386bf5b6933a9ee113956e4657fd1))
 * Remove all invocations of .first() as it is never correct unless we truly intend to grab a random record.  In all cases, .one() or much better .get() is what is really intended.  .get() won't even make a trip to the database if the object is already loaded.
   * Also prevent future invocations of .first().  Closes #418
 * fix problem caused in dataload when we reference a related object that hasn't been loaded yet during a session.flush().  In this case, the commited (and therefore incorrect or nonexistent) value of the foreign key is referenced, causing the related object to not load
 * fix a bunch of logic in the financing code, but see #425 and #353

**Comment by kbower** [comment]
_**rneville** trac comment on Nov 5, 2010, 12:54:33 PM:_

(In [[888]](../commit/2fe85a8c57c1825b6e1bba7560f7875d405cc3c9))
fixes #425 and also fixes #353

**Comment by kbower** [comment]
_**rneville** trac status change on Nov 5, 2010, 12:54:33 PM:_ **closed**
